src/chunk_index.c

```diff
@@ -907,6 +912,59 @@ chunk_index_tuple_rename(TupleInfo *ti, void *data)
 	return SCAN_DONE;
 }
 
+static ScanTupleResult
+chunk_index_tuple_adjust_meta(TupleInfo *ti, void *data)
+{
+	ChunkIndexRenameInfo *info = data;
+	bool should_free;
+	HeapTuple tuple = ts_scanner_fetch_heap_tuple(ti, false, &should_free);
+	HeapTuple new_tuple = heap_copytuple(tuple);
+	FormData_chunk_index *chunk_index = (FormData_chunk_index *) GETSTRUCT(new_tuple);
+
+	if (should_free)
+		heap_freetuple(tuple);
+
+	namestrcpy(&chunk_index->index_name, info->newname);
+	namestrcpy(&chunk_index->hypertable_index_name, info->oldname);
+
+	ts_catalog_update(ti->scanrel, new_tuple);
+	heap_freetuple(new_tuple);
+
+	return SCAN_DONE;
+}
+
+/*
+ * Adjust internal metadata after index/constraint rename
+ */
+int
+ts_chunk_index_adjust_meta(int32 chunk_id, const char *ht_index_name, const char *oldname,
```

Comment:
What is the difference between "adjust" and "rename"; especially given that this code is near identical to `ts_chunk_index_rename`?

I don't understand why we need both "adjust" and "rename", and at the very least they should share most of the code.

Reply:
adjust updates metadata only while rename has the ddl commands to actually rename the objects

Reply:
There's still the issue of code duplication. If I am not mistaken, this function is literally identical to `ts_chunk_index_rename` save for two lines of code. I think it makes sense to reuse the existing code and just call another tuple handler, or have extra options in `ChunkIndexRenameInfo` to say, e.g., "do not rename objects".